Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -2740,35 +2725,6 @@ public ConcreteSyntaxExpression(IToken tok)
   public override IEnumerable<Type> ComponentTypes => ResolvedExpression.ComponentTypes;
 }
 
-/// <summary>
-/// This class represents a piece of concrete syntax in the parse tree.  During resolution,
-/// it gets "replaced" by the statement in "ResolvedStatement".
-/// Adapted from ConcreteSyntaxStatement
-/// </summary>
-public abstract class ConcreteSyntaxStatement : Statement {
```

Comment:
Good point. But one advantage to `NestedMatchStmt.Flattened` is that it's clear that it's used for one thing, whereas `ConcreteSyntaxStatement` sounds like it could cover many instances of concrete syntax, even if it didn't, in practice.